tools/kliopt/org/jetbrains/kliopt/KliOpt.kt

```diff
@@ -172,25 +187,30 @@ class ArgParser(optionsList: List<OptionDescriptor>, argsList: List<ArgDescripto
         val name = nullArgs.firstOrNull()
         name?. let {
             argDescriptors.getValue(name).type.check(arg, name)
-            argDescriptors.getValue(name).deprecatedWarning?.let { println ("Warning: $it") }
+            argDescriptors.getValue(name).deprecated?.let { println ("${it.helpMessage(argDescriptors.getValue(name))}") }
             processedValues.getValue(name).add(arg)
             return true
         }
         return false
     }
 
-    private fun saveAsOption(descriptor: OptionDescriptor, value: String, processedValues: Map<String, MutableList<String>>) {
+    private fun saveAsOption(descriptor: OptionDescriptor, value: String,
+                             processedValues: Map<String, MutableList<String>>, useShort: Boolean = false) {
         if (!descriptor.isMultiple && !processedValues.getValue(descriptor.longName).isEmpty()) {
             printError("Option ${descriptor.longName} is used more than one time!")
         }
-        descriptor.deprecatedWarning?.let { if (processedValues.getValue(descriptor.longName).isEmpty()) println ("Warning: $it") }
+        descriptor.deprecated?.let {
+            if (processedValues.getValue(descriptor.longName).isEmpty())
```

Comment:
What this check is supposed to prevent? Reporting the same warning twice? If so, then it probably has side effects.